libtiny_tui/src/tests/mod.rs

```diff
@@ -170,3 +171,150 @@ fn small_screen_2() {
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |                                                            |
+     |osa1: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa |
+     |mentions chat.freenode.net                                  |";
+    //                                               cursor space^
+    expect_screen(screen, &tui, 60, 30);
+}
+
+///
+/// Tests scroll (text wrap on) small screen
+///
+#[test]
+fn test_scroll_text_field_wrap_true() {
+    let mut tui = TUI::new_test(30, 10);
+    tui.set_text_field_wrap_test(true);
+
+    let server = "chat.freenode.net";
+    tui.new_server_tab(server);
+    tui.set_nick(server, "osa1");
+    // switch to server tab
+    tui.next_tab();
+
+    for _ in 0..65 {
+        let event = term_input::Event::Key(Key::Char('a'));
+        tui.handle_input_event(event);
+    }
+
+    tui.draw();
+
+    let screen = "|                              |
+     |                              |
+     |                              |
+     |                              |
+     |                              |
+     |                              |
+     |                              |
+     |                              |
+     |osa1: aaaaaaaaaaaaaaaaaaaaaaa |
+     |mentions chat.freenode.net    |";
+    //                 cursor space^
+    expect_screen(screen, &tui, 30, 10);
+}
```

Comment:
Could you please make the examples smaller? No need for this large screen to test this feature. I think 10x5 screen should be enough.

It'd also be good to have some text in the messaging area while testing this, to make sure the text is rendered properly. For example, if there's a line above the input field it should be visible after text field grows one more row. Similarly if you remove text so that text field takes 1 row again the messaging area should again be resized and the live above the input field should again be right above of the input field.

If you make the screen sizes smaller it'll also take less effort to implement these tests.

Reply:
I will work on adding some more tests like one to check for the bug you mentioned a few days ago.
I think that I need the larger screen, at least to test the wrapping, because it defaults back to scrolling mode when it's too small...

Reply:
> I think that I need the larger screen, at least to test the wrapping, because it defaults back to scrolling mode when it's too small...

Hmm, right. I think you could make `OVERFLOW_WIDTH` a parameter and override it in tests. Even if we want to keep `OVERFLOW_WIDTH` constant I think you could at least make height smaller.